src/workflows/workflows.ts

```diff
@@ -303,4 +320,109 @@ export class Workflows {
       tradesApi: this.tradesApi,
     });
   }
+
+  /**
+   * IMX authorisation header functions
+   */
+  public async createProject(
+    ethSigner: EthSigner,
+    createProjectRequest: CreateProjectRequest,
+  ) {
+    const imxAuthHeaders = await generateIMXAuthorisationHeaders(ethSigner);
+
+    return this.projectsApi.createProject({
+      iMXSignature: imxAuthHeaders.signature,
+      iMXTimestamp: imxAuthHeaders.timestamp,
+      createProjectRequest,
+    });
+  }
+
+  public async getProject(ethSigner: EthSigner, id: string) {
+    const imxAuthHeaders = await generateIMXAuthorisationHeaders(ethSigner);
+
+    return this.projectsApi.getProject({
+      id,
+      iMXSignature: imxAuthHeaders.signature,
+      iMXTimestamp: imxAuthHeaders.timestamp,
+    });
+  }
+
+  public async getProjects(
+    ethSigner: EthSigner,
+    pageSize?: number,
+    cursor?: string,
+    orderBy?: string,
+    direction?: string,
+  ) {
+    const imxAuthHeaders = await generateIMXAuthorisationHeaders(ethSigner);
+
+    return this.projectsApi.getProjects({
```

Comment:
I think we previously had a user submit a PR because he couldn't figure out how to set `AxiosRequestConfig` options.
Our new interfaces encapsulate all the api methods and removed that ability to add the options for things like timeouts. Do we want to add it in?

```Typescript
export interface AxiosRequestConfig<D = any> {
  url?: string;
  method?: Method;
  baseURL?: string;
  transformRequest?: AxiosRequestTransformer | AxiosRequestTransformer[];
  transformResponse?: AxiosResponseTransformer | AxiosResponseTransformer[];
  headers?: AxiosRequestHeaders;
  params?: any;
  paramsSerializer?: (params: any) => string;
  data?: D;
  timeout?: number;
  timeoutErrorMessage?: string;
  withCredentials?: boolean;
  adapter?: AxiosAdapter;
  auth?: AxiosBasicCredentials;
  responseType?: ResponseType;
  responseEncoding?: responseEncoding | string;
  xsrfCookieName?: string;
  xsrfHeaderName?: string;
  onUploadProgress?: (progressEvent: any) => void;
  onDownloadProgress?: (progressEvent: any) => void;
  maxContentLength?: number;
  validateStatus?: ((status: number) => boolean) | null;
  maxBodyLength?: number;
  maxRedirects?: number;
  socketPath?: string | null;
  httpAgent?: any;
  httpsAgent?: any;
  proxy?: AxiosProxyConfig | false;
  cancelToken?: CancelToken;
  decompress?: boolean;
  transitional?: TransitionalOptions;
  signal?: AbortSignal;
  insecureHTTPParser?: boolean;
}
```

Reply:
Let's not add it until we understand exactly why they needed it.